### PR TITLE
Add Documentation page

### PR DESCRIPTION
**Pull request by nastyapash**

### CODE REVIEW

File: qdrant-landing/themes/qdrant-2024/layouts/partials/docs-menu.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
+{{ with (.Site.GetPage "/headless/docs-menu") }}
```
Comment (trean) [inline qdrant-landing/themes/qdrant-2024/layouts/partials/docs-menu.html:1]: We need to generate this menu from the list of documentation pages, not from markdown. 
Can you use the file `qdrant-landing/themes/qdrant/layouts/partials/docs_sidebar.html` as a reference to create it? You may replace the existing HTML with your own, but please reuse the Hugo code. You don't need to rename your file.
If you encounter any issues, please let me know.

File: qdrant-landing/themes/qdrant-2024/layouts/partials/table-contents.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{{ with (.Site.GetPage "/headless/table-contents") }}
```
Comment (trean) [inline qdrant-landing/themes/qdrant-2024/layouts/partials/table-contents.html:1]: We need to generate this from content, not from a markdown file. Here is an example: `qdrant-landing/themes/qdrant/layouts/partials/table_of_content.html`.
The `.TableOfContents` method returns an HTML list of content headings:
``` html
<nav id="TableOfContents">
  <ul>
    <li>
      <a href="/aaa">Heading h1</a>
    </li>
    <!-- and so on -->
  </ul>
</nav>
```
[Example here](https://qdrant.tech/documentation/)

**Review by trean — CHANGES_REQUESTED** [review]
Thank you for PR! 🚀 I left some change requests.

Also, do you have the latest version of the `new-design` branch here? Code snippets look like this in it.
image

And here is how they look in the current branch (seems not to functioning)
image

Normally, you don't need to do anything to add them, they already should work. Try to rebase your branch to the `new-design`.

**Comment by trean** [comment]
@nastyapash Thank you so much for the update! 🎉 

There are only two small things to finish this section:

1. The docs menu does not expand on mobile anymore, maybe some id or class got lost?
image

2. Could you add `text-overflow: ellipsis;` and indentation for nested links to the table of contents, see this page as an example https://qdrant.tech/documentation/guides/quantization/ 
image

**Comment by nastyapash** [comment]
@trean I added text trim and padding for links in the table of contents!

But opening/closing the menu works for me in the mobile version. This happens using the id #sidebar-toggle, the code is in the file qdrant-landing/themes/qdrant-2024/layouts/partials/js.html

**Comment by trean** [comment]
> @trean I added text trim and padding for links in the table of contents!
> 
> But opening/closing the menu works for me in the mobile version. This happens using the id #sidebar-toggle, the code is in the file qdrant-landing/themes/qdrant-2024/layouts/partials/js.html

I see now :-) it doesn't work because it uses jQuery. We won't have jQuery in this theme. Perhaps you have it locally somehow and this why it works on your machine. Using vanilla js will fix the problem.

The table of contents looks good, thanks!

**Comment by trean** [comment]
@nastyapash oops, one more tiny thing
Screenshot 2024-05-03 at 17 38 34

**Review by trean — APPROVED** [review]

**Pull request merged**